tests/test_stunnel.py

```diff
@@ -0,0 +1,180 @@
+"""Test module for the stunnel container image.
+
+Stunnel requires a certificate to create TLS connections. We create a self
+signed certificate for that in the :file:`tests/files/stunnel/` directory as
+follows:
+
+.. code-block:: shell-session
+
+   $ openssl genrsa -out server.key 2048
+   $ openssl req -new -key server.key -out server.csr
+   $ openssl x509 -req -days 3650 -in server.csr -signkey server.key -out server.crt -extfile v3.ext
```

Comment:
The certificate is going to expire one day, I'm not sure if there's a way to document that, otherwise if this code lives for 10+ years you get a failure to debug.

Reply:
I've added a warning